components/encryption/src/master_key/kms.rs

```diff
@@ -219,27 +231,62 @@ mod fake {
         }
 
         async fn decrypt_data_key(&self, _ciphertext: &EncryptedKey) -> Result<Vec<u8>> {
+            let fail = (|| {
+                fail_point!("kms_api_timeout_decrypt", |val| val
+                    .and_then(|x| x.parse::<bool>().ok())
+                    .unwrap_or(false));
+                false
+            })();
+            if fail {
+                return Err(CloudError::ApiTimeout(box_err!("api timeout")));
+            }
```

Comment:
```suggestion
            (|| {
                let fail = fail_point!("kms_api_timeout_decrypt", |val| val
                    .and_then(|x| x.parse::<bool>().ok())
                    .unwrap_or(false));
                if fail {
                    Err(CloudError::ApiTimeout(box_err!("api timeout")))
                } else {
                    Ok(())
                }
            })()?;
```

Reply:
I refactored the `fail_point` part to the following simplified version. Let me know if you have any thoughts or suggestions!

```rust
        fail_point!(fail_point_name, |val| {
            val.and_then(|x| x.parse::<bool>().ok())
                .filter(|&fail| fail)
                .map(|_| Err(CloudError::ApiTimeout(box_err!("api timeout"))))
                .unwrap_or(Ok(()))
        });
```